### PR TITLE
api: deprecate CreateChange using ChangeInfo (use ChangeInput instead)

### PR DESCRIPTION
**Pull request by wwade**
The API docs refer to the ChangeInput entity for create-change, see
[Documentation/rest-api-changes.html#change-input](https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#change-input).

Rename the existing CreateChange(ChangeInfo) method to
CreateChangeDeprecated and add a new CreateChange(ChangeInput)
implementation. Also added a test that ensures that none of the JSON
payload strings are empty strings (gerrit does not like those).

### CODE REVIEW

File: changes.go [hunk 1]
```diff
@@ -367,6 +367,15 @@ type AttentionSetInfo struct {
 // the edits to span newlines.
 type DiffIntralineInfo [][2]int
 
+// ChangeInput entity contains information about creating a new change.
```
Comment (andygrunwald) [inline changes.go:370]:
```suggestion
// ChangeInput entity contains information about creating a new change.
//
// Docs: https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#change-input
```

File: changes.go [hunk 2]
```diff
@@ -367,6 +367,15 @@ type AttentionSetInfo struct {
...
+type ChangeInput struct {
+	Project string `json:"project"`
+	Branch  string `json:"branch"`
+	Subject string `json:"subject"`
+	Topic   string `json:"topic,omitempty"`
+	Status  string `json:"status,omitempty"`
```
Comment (andygrunwald) [inline changes.go:376]: What do you think about to add the other properties listed in https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#change-input as well?

File: changes.go [hunk 3]
```diff
@@ -703,15 +712,38 @@ func (s *ChangesService) getCommentInfoMapSliceResponse(u string) (*map[string][
...
+		return nil, resp, err
+	}
+
+	return v, resp, err
+}
+
+// CreateChangeDeprecated creates a new change. (DEPRECATED)
+//
+// Use CreateChange instead.
+func (s *ChangesService) CreateChangeDeprecated(input *ChangeInfo) (*ChangeInfo, *Response, error) {
```
Comment (andygrunwald) [inline changes.go:746]: I think we can drop the full `CreateChangeDeprecated`.
Why? Because it will be a breaking change anyway.

Even if we introduce a `CreateChangeDeprecated`, people who update the library, need to change the call name.
If we remove `CreateChangeDeprecated` at all, they can directly change the input type from `ChangeInfo` => `ChangeInput`.

Due to the nature of (nearly) the same fields, this should be nearly no effort + they are on the right track anyway.
What do you think?

**Review by andygrunwald — CHANGES_REQUESTED** [review]
Thanks a lot for this PR.
Looks pretty solid. I made a few suggestions. Would love to know what you think.

About the change itself:
It seems to be either a bug (and works due to the same fields) or thee type was changed in one of the Gerrit versions.

**Comment by andygrunwald** [comment]
Hey @wwade,
I went ahead and made the changes

- https://github.com/wwade/go-gerrit/commit/1fe64e5283d51f4c4b09217d02fcb91af3bd9861
- https://github.com/andygrunwald/go-gerrit/commit/294d14b4b44f1fcb392ece533aed6d318fedf795
- https://github.com/andygrunwald/go-gerrit/commit/543bc02aa668ffaafed6d8afdb6f5cf9cd2f7bf6

I hope you are OK with it and it works for you.
Let me know.

File: changes.go (hunk 3 above)
Comment (wwade) [reply changes.go:748]: Yes, that makes a lot of sense. Thanks for stripping it out. Sure, I'll go ahead and add those fields.

File: changes.go (hunk 2 above)
Comment (wwade) [reply changes.go:378]: See #97
